src/core/mac/mac_frame.hpp

```diff
@@ -232,19 +232,34 @@ class TimeIe : public VendorIeHeader
      * @returns the network time, in microseconds.
      *
      */
-    uint64_t GetTime(void) const { return LittleEndian::HostSwap64(mTime); }
+    uint64_t GetTime(void) const { return LittleEndian::HostSwap64(mTimeOrTimeOffset.mTime); }
 
     /**
      * Sets the network time.
      *
-     * @param[in]  aTime  The network time.
+     * @param[in]  aTimeOffset  The network time offset.
      *
      */
-    void SetTime(uint64_t aTime) { mTime = LittleEndian::HostSwap64(aTime); }
+    void SetTimeOffset(int64_t aTimeOffset) { mTimeOrTimeOffset.mTimeOffset = aTimeOffset; }
+
+    /**
+     * Sets the network time.
+     *
+     * @param[in]  aRadioNow  Current radio time.
+     *
+     */
+    void UpdateTime(uint64_t aRadioTime)
+    {
+        mTimeOrTimeOffset.mTime = LittleEndian::HostSwap64(mTimeOrTimeOffset.mTimeOffset + aRadioTime);
+    }
 
 private:
-    uint8_t  mSequence;
-    uint64_t mTime;
+    uint8_t mSequence;
+    union
+    {
+        uint64_t mTime;       ///< The network time, only valid after UpdateTime is called.
+        int64_t  mTimeOffset; ///< The time offset relative to radio time, valid before UpdateTime is called.
+    } mTimeOrTimeOffset;
```

Comment:
The union method may cause issues if the MAC layer needs to retransmit a frame due to failures, such as No ACK, since `mTimeOffset` is overwritten.

Reply:
good catch! I'll update soon.